model/installation_request.go

```diff
@@ -90,6 +94,32 @@ func (request *CreateInstallationRequest) Validate() error {
 	return checkSpaces(request)
 }
 
+func isValidDNS(dns string) error {
+	subdomain := strings.SplitN(dns, ".", 2)[0]
+	if len(subdomain) >= 64 || len(subdomain) < 3 {
+		return errors.Errorf("DNS names must be between 3 and 64 characters, but name was %d long. DNS=%s", len(dns), dns)
+	}
```

Comment:
I believe we need to check the full DNS length here like we did before. That max character count has to apply to the full domain name.

Reply:
@gabrieljackson is this limitation an internal limitation? According to the RFC:
```
  Each element of the hostname must be from 1 to 63 characters long and
       the entire hostname, including the dots, can be at most 253
       characters long.  Valid characters for hostnames are ASCII(7) letters
       from a to z, the digits from 0 to 9, and the hyphen (-).  A hostname
       may not start with a hyphen.
```

Reply:
The RFC does make it seem like I do need to add a check to make sure the whole DNS name is <254 characters long, though.

Reply:
Okay, my recollection of the 64 limit came from what Carlos explained to me when it was first added. If we can have a subdomain with 64 characters then that is awesome!

The 254 character total length check probably won't hurt to have, but it's highly unlikely to occur given we control the rest of the domain name here. That said, I will leave it up to you to implement or not.